src/pocketmine/Player.php

```diff
@@ -2431,6 +2431,9 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 							$pk->action = AnimatePacket::ACTION_CRITICAL_HIT;
 							$pk->entityRuntimeId = $target->getId();
 							$this->server->broadcastPacket($target->getViewers(), $pk);
+							if($target instanceof Player){
```

Comment:
Or you could broadcast the packet to `$this->getViewers()` and `$this`

Reply:
@jojoe77777 no difference. In most times viewers of both attacker and target are same.

Reply:
Do all players see critical hit particles? I was under the impression that only the attacker saw them.